src/CAN/can_mapping.c

```diff
@@ -47,7 +49,30 @@ float canmapping_extract_value(uint64_t raw_data, const CANMapping *mapping)
                         break;
                 }
         }
-        return (float)raw_value;
+
+        /* convert type */
+        float extracted_value = 0;
+        switch (mapping->type) {
+        	case CANMappingType_IEEE754:
+        		extracted_value = *((float*)&raw_value);
+        		break;
+        	case CANMappingType_signed:
+        		if (mapping->length <= 8) {
+        			extracted_value = *((int8_t*)&raw_value);
```

Comment:
@stieg  this uses C type punning to coerce the unsigned value into the target type. The cast to the final (normalized) float type is implicit; I agree it could be clearer - e.g.

extracted_value = (float)*((int16_t*)&raw_value);